Introducing WIF in Secret Sync for Azure Destination (#1816)
hashicorp/web-unified-docs: content/vault/v2.x (rc)/content/api-docs/system/secrets-sync.mdx

- `client_id` `(string: <required>)` - Client ID of an Azure app registration.

> - `client_secret` `(string: "")` - Client secret of an Azure app registration. Required for client secret authentication. Mutually exclusive with `identity_token_audience`. If omitted, authentication falls back to workload identity federation (WIF) when `identity_token_audience` is set, or Azure managed identity when neither is provided.

Reviewer: ```suggestion
- `client_secret` `(string: "")` - Client secret of an Azure app registration.
  Required for client secret authentication and mutually exclusive with
  `identity_token_audience`. If you omit `client_secret`, Vault falls back to
  alternative authentication methods:
  - Set `identity_token_audience` to fall back to workload identity federation
    (WIF).
  - Leave `client_secret` and `identity_token_audience` unset to fall back to
    Azure managed identity.
```
Style correction: write in active voice

- `tenant_id` `(string: <required>)` - ID of the target Azure tenant.

- `identity_token_audience` `(string: "")` - Audience claim value for identity tokens used in workload identity federation (WIF) authentication. The value must match the allowed audiences configured for the federated identity credential in Azure. Required for WIF authentication. Mutually exclusive with `client_secret`.

> - `identity_token_key` `(string: "")` - Name of a configured identity token key to use for signing identity tokens for WIF authentication. [Create the key](/vault/api-docs/secret/identity/tokens#create-a-named-key) before configuring this destination. If omitted, the [default](/vault/docs/concepts/oidc-provider#keys) identity token signing key is used.

Reviewer: ```suggestion
- `identity_token_key` `(string: "")` - Name of a configured identity token key to use for signing identity tokens for WIF authentication. [Create the key](/vault/api-docs/secret/identity/tokens#create-a-named-key) before configuring the identity token key. Leave `identity_token_key` unset to use the [default](/vault/docs/concepts/oidc-provider#keys) identity token signing key.
```
Style correction: write in active voice, avoid "this" as a pronoun
